ouroboros-consensus/src/Ouroboros/Consensus/Node/Types.hs

```diff
@@ -0,0 +1,45 @@
+{-# LANGUAGE DeriveAnyClass     #-}
+{-# LANGUAGE DeriveGeneric      #-}
+{-# LANGUAGE DerivingStrategies #-}
+
+module Ouroboros.Consensus.Node.Types (
+  CandidateDecline (..),
+  CandidateFragment (..),
+  CandidateFingerprint (..),
+  ) where
+
+import           GHC.Generics (Generic)
+import           NoThunks.Class
+
+import           Ouroboros.Network.AnchoredFragment (AnchoredFragment)
+
+import           Ouroboros.Consensus.Block (Point)
+
+-- | What ChainSync provides per peer to BlockFetch.
+data CandidateFragment header = CandidateFragment {
+    candidateIsLowDensity :: !Bool
+  , candidateChain        :: !(AnchoredFragment header)
+  }
+  deriving stock    (Generic)
+  deriving anyclass (NoThunks)
+
+-- | A reason that we might choose to ignore a 'CandidateFragment'.
+data CandidateDecline =
+    -- | The 'candidateChain' is not better than our current selected chain.
+    DeclineNotPlausible
+    -- | The candidate is empty (which should only happen if
```

Comment:
"has no new blocks -- all are already on our current chain"    eg rename to `DeclineAlreadyDownloaded`